### PR TITLE
fix: use Patch API for infra-client

### LINKED ISSUE #2995: Move to Patch API on infra-client
*Description*:
Follow up from #2807, which is still happening in the latest version (v1.0.0).  I noticed this behavior because the EG is utilizing the Update API instead of the Patch API in infra-client.
https://github.com/envoyproxy/gateway/blob/c655811da9310fc474164af8d97f929dae248cff/internal/infrastructure/kubernetes/infra_client.go#L40-L45

As a result, instead of only updating the changed field, it turned out to re-applies all fields causing the field to revert to its original value, instead of updating what changed. For example on the Deployment object, if the replicas field is set to null, it will be reverted to null.

So, I tried in my local to use Patch API, and it worked as expected.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by ardikabs**
**What type of PR is this?**
Instead of utilizing the Update API for the infra-client, this PR changed it to the Patch API with ApplyPatch.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2995

### CODE REVIEW

File: internal/infrastructure/kubernetes/infra_client.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ func (cli *InfraClient) CreateOrUpdate(ctx context.Context, key client.ObjectKey
 			// just perform an update for now.
 			if updateChecker() {
 				specific.SetUID(current.GetUID())
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/infra_client.go:41]: is this `SetUID` still needed ?

File: charts/gateway-helm/templates/infra-manager-rbac.yaml [hunk 2]
```diff
@@ -34,6 +36,7 @@ rules:
   - get
   - update
```
[the remark this answers is not in the document]
Comment (ardikabs) [reply charts/gateway-helm/templates/infra-manager-rbac.yaml]: changed

File: internal/infrastructure/kubernetes/infra_client.go (hunk 1 above)
Comment (ardikabs) [reply internal/infrastructure/kubernetes/infra_client.go]: yup, it's no longer needed.

**Comment by ardikabs** [comment]
> ## [Codecov](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy) Report
> Attention: Patch coverage is `50.00000%` with `1 lines` in your changes are missing coverage. Please review.
> 
> > Project coverage is 64.52%. Comparing base [(`a11aa70`)](https://app.codecov.io/gh/envoyproxy/gateway/commit/a11aa70ddf78c058b38400044be5c4a5e2944fc4?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy) to head [(`7cf0195`)](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy).
> 
> [Files](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy)	Patch %	Lines
> [internal/infrastructure/kubernetes/infra_client.go](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy#diff-aW50ZXJuYWwvaW5mcmFzdHJ1Y3R1cmUva3ViZXJuZXRlcy9pbmZyYV9jbGllbnQuZ28=)	50.00%	[0 Missing and 1 partial ⚠️ ](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy)
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy).

Do you have any suggestions for this, @arkodg ? I'm looking to achieve 60% coverage, but there's no corresponding unit test file for the changed code.

**Comment by zirain** [comment]
> > ## [Codecov](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy) Report
> > Attention: Patch coverage is `50.00000%` with `1 lines` in your changes are missing coverage. Please review.
> > > Project coverage is 64.52%. Comparing base [(`a11aa70`)](https://app.codecov.io/gh/envoyproxy/gateway/commit/a11aa70ddf78c058b38400044be5c4a5e2944fc4?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy) to head [(`7cf0195`)](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy).
> > 
> > 
> > [Files](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy)	Patch %	Lines
> > [internal/infrastructure/kubernetes/infra_client.go](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy#diff-aW50ZXJuYWwvaW5mcmFzdHJ1Y3R1cmUva3ViZXJuZXRlcy9pbmZyYV9jbGllbnQuZ28=)	50.00%	[0 Missing and 1 partial ⚠️ ](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy)
> > Additional details and impacted files
> > [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/envoyproxy/gateway/pull/3034?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=envoyproxy).
> 
> Do you have any suggestions for this, @arkodg ? I'm looking to achieve 60% coverage, but there's no corresponding unit test file for the changed code.

coverage is not a must for a PR.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**